lib/src/components/layout/accordion.dart

```diff
@@ -20,39 +20,149 @@ class _AccordionState extends State<Accordion> {
+                        ? item._applyTheme(accTheme)
+                        : item;
+                  }),
+                      ShadcnAnimatedContainer(
+                        duration: kDefaultDuration,
+                        color: theme.colorScheme.muted,
+                        height: accTheme.dividerHeight * scaling,
+                      )),
+                  const Divider(),
+                ]),
+          ),
         ));
   }
 }
 
+/// {@template accordion_theme}
+/// Styling options for [AccordionItem], [AccordionTrigger] and [Accordion].
+/// {@endtemplate}
+class AccordionTheme {
+  /// Duration of the collapse/expand animation.
+  final Duration duration;
+
+  /// Curve of the animation (played when expanding).
+  final Curve curve;
+
+  /// Reverse curve of the animation (played when collapsing).
+  final Curve reverseCurve;
+
+  /// The gap between the trigger and the content (or other triggers if collapsed).
+  ///
+  /// The padding is applied to the top and bottom of the trigger.
+  final double padding;
+
+  /// The gap between the trigger text and the icon.
+  final double iconGap;
+
+  /// The height of the divider between each item.
+  final double dividerHeight;
+
+  /// {@macro accordion_theme}
+  const AccordionTheme({
+    this.duration = const Duration(milliseconds: 200),
+    this.curve = Curves.easeIn,
+    this.reverseCurve = Curves.easeOut,
+    this.padding = 16,
+    this.iconGap = 18,
+    this.dividerHeight = 1,
+  });
+
+  /// Creates a copy of this theme and replaces the given properties.
+  ///
+  /// {@macro accordion_theme}
+  AccordionTheme copyWith({
+    Duration? duration,
+    Curve? curve,
+    Curve? reverseCurve,
+    double? padding,
+    double? iconGap,
+    double? dividerHeight,
+  }) {
+    return AccordionTheme(
+      duration: duration ?? this.duration,
+      curve: curve ?? this.curve,
+      reverseCurve: reverseCurve ?? this.reverseCurve,
+      padding: padding ?? this.padding,
+      iconGap: iconGap ?? this.iconGap,
+      dividerHeight: dividerHeight ?? this.dividerHeight,
+    );
+  }
+
+  @override
+  bool operator ==(Object other) =>
+      other is AccordionTheme &&
+      duration == other.duration &&
+      curve == other.curve &&
+      reverseCurve == other.reverseCurve &&
+      padding == other.padding &&
+      iconGap == other.iconGap &&
+      dividerHeight == other.dividerHeight;
+
+  @override
+  int get hashCode => Object.hash(
+        duration,
+        curve,
+        reverseCurve,
+        padding,
+        iconGap,
+        dividerHeight,
+      );
+
+  @override
+  String toString() {
+    return 'AccordionTheme(duration: $duration, curve: $curve, reverseCurve: $reverseCurve, padding: $padding, iconGap: $iconGap, dividerHeight: $dividerHeight)';
+  }
+}
+
 class AccordionItem extends StatefulWidget {
   final Widget trigger;
   final Widget content;
   final bool expanded;
+  final AccordionTheme theme;
```

Comment:
Do not pass the theme to the component, this is why we have `ComponentTheme` class